gammapy/data/tests/test_observations.py

```diff
@@ -38,6 +34,26 @@ def test_data_store_observation(data_store):
     assert_skycoord_allclose(obs.target_radec, c)
 
 
+@requires_data("gammapy-extra")
+def test_missing_gti(data_store):
+    """This tests the case when a GTI table is missing in the HDU index file.
+
+    For backward compatibility we create a GTI table on-the-fly by means of the Obs Index file
+    TODO: This method can be removed once GTI tables are explicitly required in Gammapy.
```

Comment:
@dcfidalgo - I'll remove this test already now. I've had very bad experiences with tests that modify fixture objects, and then depending on test execution order other tests broke and it was very hard to debug. I also don't like to have such complex tests where state is backed up and restored.
Overall this is an edge case that should go away soon, I think it's OK to leave this one line untested.